include/libtorrent/kademlia/node.hpp

```diff
@@ -75,22 +76,25 @@ class announce_observer : public observer
 	void reply(msg const&) { flags |= flag_done; }
 };
 
-struct udp_socket_interface
+struct socket_manager
 {
 	virtual bool has_quota() = 0;
-	virtual bool send_packet(entry& e, udp::endpoint const& addr) = 0;
+	virtual bool send_packet(aux::session_listen_socket* s, entry& e, udp::endpoint const& addr) = 0;
 protected:
-	~udp_socket_interface() {}
+	~socket_manager() {}
```

Comment:
I don't actually have a strong opinion here, if you think ``{}`` is better, that's fine (but you changed a few other ``{}`` -> ``= default`` I noticed)